pkg/controller/operators/openshift/helpers.go

```diff
@@ -178,57 +205,37 @@ func maxOpenShiftVersion(csv *operatorsv1alpha1.ClusterServiceVersion) (*semver.
 		return nil, err
 	}
 
-	// Take the highest semver if there's more than one max version specified
+	// Take the highest semver if there's more than one max version specified.
+	// Note: duplicate versions are not a problem because the intent to block a specific X.Y-stream is still clear.
```

Comment:
I feel like an argument could be made for accepting the lowest semver. If we accept highest, would that mean we'd technically be violating all the lower semver constraints?

Reply:
I can see that argument.

I was thinking that it would be more common for authors to add higher versions, as they test compatibility with new OpenShift releases, than it would be to add lower versions over time.

Another route could be to require the max version property to be unique; i.e. be less tolerant of sloppy property maintenance.

Reply:
Ideally, I think enforcing uniqueness is the right answer, mainly because that removes all ambiguity.

I guess the thing to think about is where the properties come from and if we think there's a use case for them being additive like you suggest.

IMO, with an eye towards declarative config, and additive style flow sounds more imperative and probably not something we want to encourage or reinforce, so I think I'd vote for a uniqueness requirement on this property.

Reply:
okay, you've convinced me that enforcing uniqueness is the best choice here. Will update.

/hold